include/real/exact_number.hpp

```diff
@@ -264,6 +272,116 @@ namespace boost {
                 this->normalize();
             }
 
+            /**     KARATSUBA MULTIPLICATION
+             *  @brief: multiplies (*this) with other using karatsuba multiplication algorithm
+             *  @param: other: an exact_number to be multiplied with (*this)
+             *  @param: base: base of the numbers being multiplied
+             *  @author: Kishan Shukla
+             */
+
+            void karatsuba_multiplication (
+                    exact_number<T> &other, 
+                    const T base = (std::numeric_limits<T>::max() / 4) * 2
+            ) {
+
+                while (this->exponent > (int)this->digits.size()) {
+                    this->digits.push_back(0);
+                }
+
+                while (other.exponent > (int)other.digits.size()) {
+                    other.digits.push_back(0);
+                }
+
+                // this --- a, other --- b
+
+            	int a_size = this->digits.size();
+            	int b_size = other.digits.size();
+
+                const int max_length = std::max(a_size, b_size);
+
+                if (max_length <= KARATSUBA_THRESHOLD) {
```

Comment:
I think that redirecting to the standard multiplication inside karatsuba multiplication can not be very clear. It may be better to place all the karatsuba code until this line in the  multiply_vector function and call the karatsuba or the standard multiplication (depending on the threshold) from there.

Reply:
This is the base case of the recursive algorithm. Even if we do what you are saying, still we are going to keep this in the karatsuba function. This provides significant optimization.